File: test/fixtures/ruby-simple/lib/service.rb
require_relative 'user'
require 'sinatra'

module MyApp
  class UserService
    def initialize
      @users = []
    end

    def all
      @users
    end

    def find(id)
      @users.find { |u| u.id == id }
    end

    def create(name, email)
      user = User.new(@users.length + 1, name, email)
      @users << user
      user
    end
  end
end

File: test/fixtures/rust-simple/src/service.rs
use crate::models::{User, Status, validate_email};

pub struct UserService;

impl UserService {
    pub fn create(name: String, email: String) -> User {
        User { id: 1, name, email }
    }

    pub fn get_by_id(id: u64) -> Option<User> {
        None
    }
}

pub trait Repository<T> {
    fn find_by_id(&self, id: u64) -> Option<T>;
    fn save(&self, item: &T) -> bool;
}
